hack/dailybuild/dailybuild.go

```diff
@@ -0,0 +1,232 @@
+// Copyright 2021 VMware Tanzu Community Edition contributors. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/gorilla/feeds"
+)
+
+const (
+	projectID  string = "vmware-cna"
+	bucketName string = "tce-cli-plugins-staging"
+	gcpFile    string = "gcptokenfile.json"
+	rssFile    string = "../../rss.xml"
+)
+
+type ClientUploader struct {
+	cl         *storage.Client
+	projectID  string
+	bucketName string
+	uploadPath string
+}
+
+// UploadFile uploads an object
+func (c *ClientUploader) UploadFile(filename string) error {
+	ctx := context.Background()
+
+	ctx, cancel := context.WithTimeout(ctx, time.Second*1200)
+	defer cancel()
+
+	// open the src file
+	uploadFile := filepath.Join("..", "..", "build", filename)
+	file, err := os.OpenFile(uploadFile, os.O_RDWR, 0755)
+	if err != nil {
+		fmt.Printf("OpenFile failed: %v\n", err)
+		return err
+	}
+
+	// Upload an object with storage.Writer.
+	wc := c.cl.Bucket(c.bucketName).Object(c.uploadPath + filename).NewWriter(ctx)
+	if _, err := io.Copy(wc, file); err != nil {
+		fmt.Printf("io.Copy: %v\n", err)
+		return err
+	}
+	if err := wc.Close(); err != nil {
+		fmt.Printf("Writer.Close: %v\n", err)
+	}
+
+	return nil
+}
+
+func upload(platform, arch, tag, extension string) (string, error) {
+	os.Setenv("GOOGLE_APPLICATION_CREDENTIALS", gcpFile)
+
+	client, err := storage.NewClient(context.Background())
+	if err != nil {
+		fmt.Printf("Failed to create client: %v\n", err)
+		return "", err
+	}
+
+	assetFilename := fmt.Sprintf("tce-%s-%s-%s.%s", platform, arch, tag, extension)
+	uploadPathTmp := filepath.Join("build-daily", time.Now().Format("2006-01-02"))
+	uploadPath := fmt.Sprintf("%s/", uploadPathTmp)
+
+	uploader := &ClientUploader{
+		cl:         client,
+		bucketName: bucketName,
+		projectID:  projectID,
+		uploadPath: uploadPath,
+	}
+
+	err = uploader.UploadFile(assetFilename)
+	if err != nil {
+		fmt.Printf("UploadFile failed: %v\n", err)
+		return "", err
+	}
+
+	urlUpload := fmt.Sprintf("https://storage.googleapis.com/%s/%s%s", bucketName, uploadPath, assetFilename)
+	return urlUpload, nil
+}
+
+func uploadAndUpdateRss(tag, gcpToken string) error {
+	// upload
+	err := os.WriteFile(gcpFile, []byte(gcpToken), 0644)
+	if err != nil {
+		fmt.Printf("WriteFile failed: %v\n", err)
+		return err
+	}
+
+	urlLinuxAmd64, err := upload("linux", "amd64", tag, "tar.gz")
```

Comment:
Something to think about as a future optimization... I wonder if we can plumb through `ENVS` from the Makefile so this automatically picks up all platforms we're building for. Then in here, we can just loop through each `ENV` entry to process it.

Reply:
ahhhhh that's a good idea. will need to think about hooking that up later.

Reply:
one problem we currently have is that we are building darwin arm64 but do we want to include it in the list of the daily build. at some point when we do support it we definitely will... but listing it right now might lead people to think it's a support platform/arch